lib/nrf_modem_lib/trace_backends/uart/Kconfig

```diff
@@ -8,18 +8,37 @@
 choice NRF_MODEM_LIB_TRACE_BACKEND
 
 config NRF_MODEM_LIB_TRACE_BACKEND_UART
-	bool "UARTE1"
-	# Modem tracing over UART use the UARTE1 as a dedicated peripheral.
-	# This enables the UARTE1 peripheral and includes nrfx UARTE driver.
-	select NRFX_UARTE1
+	bool "UART"
 
 if NRF_MODEM_LIB_TRACE_BACKEND_UART
 
+choice NRF_MODEM_LIB_TRACE_BACKEND_UART_VERSION
+	prompt "UART trace backend version"
+	default NRF_MODEM_LIB_TRACE_BACKEND_UART_ASYNC
```

Comment:
The reason for using the deprecated backend for default is that there is a lot of documentation and guides around that needs to be updated for the new trace backend (building with snippet instead of setting a Kconfig). So the plan is to keep the deprecated one as default for now, so that these guides will still work, but give a warning to the user. Then we will change the default and remove the old one in a later NCS version.

Reply:
Did you see the question about defaulting to the new uart backend when there is a chosen node for `nordic,modem-trace-uart`?

Reply:
No, that went past me :/ Added it now, thanks!